libs/barista-components/table/src/selection/selector.spec.ts

```diff
@@ -0,0 +1,328 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* tslint:disable:no-unused-variable */
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { Component, ViewChild, Predicate } from '@angular/core';
+import { DtTableModule } from '@dynatrace/barista-components/table';
+import { CommonModule } from '@angular/common';
+import { DtIconModule } from '@dynatrace/barista-components/icon';
+import { DtLoadingDistractorModule } from '@dynatrace/barista-components/loading-distractor';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { DtFormattersModule } from '@dynatrace/barista-components/formatters';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { By } from '@angular/platform-browser';
+import { DtCheckbox } from '@dynatrace/barista-components/checkbox';
+import { DtTableSelection, DT_TABLE_SELECTION_CONFIG } from './selection';
+
+describe('DtTableSelection', () => {
+  let component: DtTableSelectionComponentForTesting;
+  let fixture: ComponentFixture<DtTableSelectionComponentForTesting>;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        CommonModule,
+        DtTableModule,
+        DtIconModule.forRoot({ svgIconLocation: `{{name}}.svg` }),
+        DtLoadingDistractorModule,
+        NoopAnimationsModule,
+        DtFormattersModule,
+        HttpClientTestingModule,
+      ],
+      // The providers are necessary here - because otherwise the TestBed does not allow
+      // overriding the providers in a different setup below
+      // The token needs to be already available initially
+      providers: [
+        {
+          provide: DT_TABLE_SELECTION_CONFIG,
+          useValue: {},
+        },
+      ],
+      declarations: [DtTableSelectionComponentForTesting],
+    });
+  });
+
+  describe('without limit', () => {
+    beforeEach(() => {
+      TestBed.compileComponents();
+      fixture = TestBed.createComponent(DtTableSelectionComponentForTesting);
+      component = fixture.componentInstance;
+      fixture.detectChanges();
```

Comment:
Nitpick:
You could use the provided helper function for this. It creates the component and calls the first detect changes.
Add the import
```
import { createComponent } from '@dynatrace/testing/browser';
```

```suggestion
      fixture = createComponent(DtTableSelectionComponentForTesting);
      component = fixture.componentInstance;
```